libgadget/forcetree.c

```diff
@@ -78,13 +78,18 @@ force_tree_eh_slots_fork(EIBase * event, void * userdata)
     EISlotsFork * ev = (EISlotsFork*) event;
     int parent = ev->parent;
     int child = ev->child;
-    int no;
     ForceTree * tree = (ForceTree * ) userdata;
-    no = tree->Nextnode[parent];
-    tree->Nextnode[parent] = child;
-    tree->Nextnode[child] = no;
-    tree->Father[child] = tree->Father[parent];
-
+    int no = force_get_father(parent, tree);
+    struct NODE * nop = &tree->Nodes[no];
+    /* FIXME: We lose particles if the node is full.
+     * At the moment this does not matter, because
+     * the only new particles are stars, which do not
+     * participate in the SPH tree walk.*/
+    if(nop->s.noccupied < NMAXCHILD) {
+       nop->s.suns[nop->s.noccupied-1] = child;
+        nop->s.noccupied++;
+    }
```

Comment:
From your description, sounds like we shall crash here; at least print something?

Reply:
We don't crash, we just don't add the particle to the tree. This has no negative consequences at this time, because we don't do anything with the newly created particle anyway! Perhaps a message at DEBUG? What should it say?